assets/training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -250,10 +258,26 @@ def _get_model_properties(self):
 
     def download_model(self):
         """Download a Hugging face model and return details."""
+        if self._token:
+            # Command to set the 'store' credential helper as default, as no helper is defined by default.
+            cmd = "git config --global credential.helper store"
+            exit_code, stdout = run_command(cmd)
+            if exit_code != 0:
+                raise AzureMLException._with_error(
+                    AzureMLError.create(GITConfigError, error=stdout)
+                )
+            try:
+                login(token=self._token, add_to_git_credential=True)
+            except Exception as ex:
+                raise AzureMLException._with_error(
+                    AzureMLError.create(HFAuthenticationError, error=ex)
+                )
+
         if self.model_info:
             download_details = self._download()
             model_props = self._get_model_properties()
             model_props.update(download_details)
+            logout()
```

Comment:
will logout also throw any exception? Please catch it as well

Reply:
Added